TrackletAlgorithm/MemoryTemplate.h

```diff
@@ -130,25 +130,35 @@ class MemoryTemplate
 
   void print_mem(BunchXingT bx) const
   {
-	for (int i = 0; i <  nentries_[bx]; ++i) {
+	for (unsigned int i = 0; i <  nentries_[bx]; ++i) {
 	  std::cout << bx << " " << i << " ";
 	  print_entry(bx,i);
 	}
   }
 
   void print_mem() const
   {
-	for (int ibx = 0; ibx < (1<<NBIT_BX); ++ibx) {
-	  for (int i = 0; i < nentries_[ibx]; ++i) {
+	for (unsigned int ibx = 0; ibx < (1<<NBIT_BX); ++ibx) {
+	  for (unsigned int i = 0; i < nentries_[ibx]; ++i) {
 		std::cout << ibx << " " << i << " ";
 		print_entry(ibx,i);
 	  }
 	}
   }
 
   static constexpr int getWidth() {return DataType::getWidth();}
-  
-#endif
+
+  static std::string decodeToBits(unsigned int field, unsigned int size) {
+    unsigned int valtmp = field;
+    std::string str = "";
+    for(unsigned int i=0; i< size; i++) {
+      str = ((valtmp & 1) ? "1" : "0") + str;
+      valtmp >>= 1;
+    }
+    return str;
+  }
```

Comment:
@carriganm95 Can you remove the versions of this function in the `MemoryTemplate`* classes?

Reply:
removed